packages/node/src/algorand/api.algorand.ts

```diff
@@ -0,0 +1,254 @@
+import axios from 'axios';
+import { camelCaseObjectKey } from './utils.algorand';
+
+const logger = getLogger('api.algorand');
+
+export class AlgorandApi {
+  private genesisHash: string;
+  private chain: string;
+  api: Indexer;
+  private blockCache: AlgorandBlock[];
+  private paginationLimit = 10000;
+
+  constructor(private endpoint: string, private token?: string | TokenHeader) {
+    this.token = token ?? '';
+    this.blockCache = [];
+  }
+
+  async init(): Promise<void> {
+    // get genesisHash in block
+    const urlEndpoint = new URL(this.endpoint);
+    const baseServer = `${urlEndpoint.protocol}//${urlEndpoint.host}${urlEndpoint.pathname}`;
+
+    this.api = new algosdk.Indexer(this.token, baseServer, urlEndpoint.port);
+
+    try {
+      const block = await this.api.lookupBlock(1).do();
+
+      this.genesisHash = block['genesis-hash'] ?? '';
+      this.chain = block['genesis-id'] ?? '';
+    } catch (e) {
+      logger.error(e);
+      process.exit(1);
+    }
+  }
+
+  async getBlockByHeight(height: number): Promise<AlgorandBlock> {
+    try {
+      const blockInfo = await this.api.lookupBlock(height).do();
+      return camelCaseObjectKey(blockInfo);
+    } catch (error) {
+      if (error.message.includes('Max transactions limit exceeded')) {
+        logger.warn('Max transactions limit exceeded, paging transactions');
+
+        return this.combinePaginateBlock(height);
+      }
+      logger.error(`failed to fetch Block at round ${height}`);
+      throw error;
+    }
+  }
+
+  async getHeaderOnly(block: number): Promise<any> {
+    try {
+      const result = (
+        await axios({
+          params: {
+            'header-only': true,
+          },
+          method: 'get',
+          url: `/v2/blocks/${block}`,
+          baseURL: this.endpoint,
+        })
+      ).data;
+      return result;
+    } catch (e) {
+      logger.error('Failed to fetch round header', e);
+      throw e;
+    }
+  }
+
+  async paginatedTransactions(
+    blockHeight: number,
+    combinedTx: any[] = [],
+    nextToken?: string,
+  ): Promise<any> {
+    try {
+      const result: { transactions: any[] } = (
+        await axios({
+          params: {
+            round: blockHeight,
+            limit: this.paginationLimit,
+            next: nextToken && nextToken,
+          },
+          method: 'get',
+          url: `/v2/transactions`,
+          baseURL: this.endpoint,
+        })
+      ).data;
+      /*
+      Maximum number of results to return. There could be additional pages even if the limit is not reached.
+      https://developer.algorand.org/docs/rest-apis/indexer/#get-v2transactions
+
+      Hence, the condition below
+       */
+      if (result.transactions.length > 0) {
+        combinedTx.push(result.transactions);
+
+        return await this.paginatedTransactions(
+          blockHeight,
+          combinedTx,
+          result['next-token'],
+        );
+      }
+      return [].concat(...combinedTx);
```

Comment:
Something like this:

```suggestion
      if (result.transactions.length > 0) {

        const nextpage = await this.paginatedTransactions(
          blockHeight,
          combinedTx,
          result['next-token'],
        );
        
        return result.transactions.concat(nextpage);
      }
      return results.transactions;
```